File: clientlibs/js/src/functions/setGroupMembership.ts
import { UpGradeClientInterfaces, UpGradeClientEnums } from '../types';
import fetchDataService from '../common/fetchDataService';

export default async function setGroupMembership(
  customHttpClient: UpGradeClientInterfaces.ICustomHttpClient,
  url: string,
  userId: string,
  token: string,
  clientSessionId: string,
  group: Record<string, Array<string>>
): Promise<UpGradeClientInterfaces.IUser> {
  const response = await fetchDataService(
    customHttpClient,
    url,
    token,
    clientSessionId,
    { id: userId, group: group },
    UpGradeClientEnums.REQUEST_TYPES.PATCH
  );
  if (response.status) {
    return response.data;
  } else {
    throw new Error(JSON.stringify(response.message));
  }
}

File: clientlibs/csharp/src/functions/SetGroupMembership.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Enums;
using Interfaces;
using Newtonsoft.Json;

public static class GroupClass
{
  public static async Task<Group> SetGroupMembership(
    string url,
    string userId,
    string token,
    string clientSessionId,
    Dictionary<string, string[]> group)
  {
    var response = await FetchDataServiceClass.FetchDataService(
        url,
        token,
        clientSessionId,
        new { id = userId, group },
        REQUEST_TYPES.PATCH
    );
    if (response.status)
    {
      var responseData = JsonConvert.DeserializeObject<Group>(response.data.ToString());
      return responseData as Group;
    }
    else
    {
      throw new Exception(JsonConvert.SerializeObject(response.message));
    }
  }
}

